tracee-ebpf/tracee/network_protocols/kernel_functions_based_net_events.go

```diff
@@ -0,0 +1,67 @@
+package network_protocols
+
+import (
+	"bytes"
+	"encoding/binary"
+	"github.com/aquasecurity/tracee/pkg/external"
+	"github.com/aquasecurity/tracee/pkg/processContext"
+)
+
+// FunctionBasedNetEvents are net events originated from kernel functions (kprobes) rather than from the tc (tc_probes).
+type FunctionBasedPacket struct {
+	LocalIP     [16]byte
+	RemoteIP    [16]byte
+	LocalPort   uint16
+	RemotePort  uint16
+	Protocol    uint8
+	_           [3]byte //padding
+	TcpOldState uint32
+	TcpNewState uint32
+	_           [4]byte //padding
+	SockPtr     uint64
+}
+
+func FunctionBasedNetEventHandler(buffer *bytes.Buffer, evtMeta EventMeta, ctx processContext.ProcessCtx, eventName string) external.Event {
+	var evt external.Event
+	functionBasedEventPacket, err := ParseFunctionBasedPacketMetaData(buffer)
+	if err != nil {
+		return evt
+	}
+	evt = CreateNetEvent(evtMeta, ctx)
+	CreateFunctionBasedPacketMetadataArgs(&evt, functionBasedEventPacket)
+	evt.EventName = eventName
+	return evt
+}
+
+// parsing the PacketMeta struct from bytes.buffer
+func ParseFunctionBasedPacketMetaData(payload *bytes.Buffer) (FunctionBasedPacket, error) {
+	var pktMetaData FunctionBasedPacket
+	err := binary.Read(payload, binary.LittleEndian, &pktMetaData)
+	if err != nil {
+		return pktMetaData, err
+	}
+
+	return pktMetaData, nil
+}
+
+func CreateFunctionBasedPacketMetadataArgs(event *external.Event, packet FunctionBasedPacket) {
+	eventArgs := make([]external.Argument, 0, 0)
+	arg := external.FunctionBasedPacket{}
+	arg.LocalIP = parseIP(packet.LocalIP)
```

Comment:
you can use `external.PktMeta` for the first arg (src ip, src port, dst ip, dst port, protocol). then you can break the rest into args:
tcp_new_state
tcp_old_state
sock_ptr

Reply:
well yes and no, because of in the function based net events (formerly we called it `debug_net_events`) we used a similar struct with different name members.
Like `src ip `changed to `local ip`,  and `src port`, to `loacl port` etc...

I can change it if  you think it ok, let me know what you think

Reply:
ohh i see.. ok so don't change it